### PR TITLE
[subset] Prune GDEF.MarkGlyphSetsDef

### PR DESCRIPTION
**Pull request by khaledhosny**
Remove empty sets and fixup Lookup.MarkFilteringSet to use the new indices.

### CODE REVIEW

**Comment by khaledhosny** [comment]
Needs a test.

**Review by anthrotype — APPROVED** [review]
LGTM!

**Comment by anthrotype** [comment]
does hb-subset do the same? if it, I suppose it should also be updated /cc @qxliu76 @garretrieger @behdad

**Comment by khaledhosny** [comment]
> does hb-subset do the same? if not, I suppose it should also be updated

Looks like hb-subset does not prune them.

**Comment by anthrotype** [comment]
> Looks like hb-subset does not prune them.

maybe file an issue upstream in harfbuzz/harfbuzz

**Comment by khaledhosny** [comment]
> > Looks like hb-subset does not prune them.
> 
> thanks for checking, maybe file an issue upstream in harfbuzz/harfbuzz

https://github.com/harfbuzz/harfbuzz/issues/4239

File: Lib/fontTools/subset/__init__.py [hunk 1]
```diff
@@ -2091,17 +2097,14 @@ def subset_glyphs(self, s):
...
                 coverage.subset(glyphs)
 
-        # TODO: The following is disabled. If enabling, we need to go fixup all
-        # lookups that use MarkFilteringSet and map their set.
-        # indices = table.MarkGlyphSetsDef.Coverage = \
-        #   [c for c in table.MarkGlyphSetsDef.Coverage if c.glyphs]
-        # TODO: The following is disabled, as ots doesn't like it. Phew...
-        # https://github.com/khaledhosny/ots/issues/172
-        # table.MarkGlyphSetsDef.Coverage = [c if c.glyphs else None for c in table.MarkGlyphSetsDef.Coverage]
+        s.used_mark_sets = [i for i, c in enumerate(markGlyphSets.Coverage) if c.glyphs]
```
Comment (behdad) [inline Lib/fontTools/subset/__init__.py:2105]: This introduces a dependency on the order the tables are subset. We don't enforce that right now.

Comment (khaledhosny) [reply Lib/fontTools/subset/__init__.py:2105]: I thought `_sort_tables` would sort the tables alphabatically, so `GDEF` always comes first, but reading again it seem to leave the other tables in their original order. May be I should add `GDEF`, `GSUB` and `GPOS` their?
https://github.com/fonttools/fonttools/blob/a537f97b44118e7a14cc6e507701e9fca95b81e5/Lib/fontTools/subset/__init__.py#L3498C4-L3502

Comment (behdad) [reply Lib/fontTools/subset/__init__.py:2105]: Sounds right.

Comment (khaledhosny) [reply Lib/fontTools/subset/__init__.py:2105]: https://github.com/fonttools/fonttools/pull/3118

**Pull request merged**